### PR TITLE
feat(): option to save scroll buffer

### PR DESCRIPTION
**Pull request by marhi**
Context menu option to save terminal scroll buffer into a file.

### CODE REVIEW

File: src/app/components/window-terminal/window-terminal.component.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ export class WindowTerminalComponent implements OnInit, OnDestroy {
       label: "Paste",
       click: () => this.paste()
     }));
+    this.ctxMenu.append(new MenuItem({
+      label: "Save Buffer",
```
Comment (jkuri) [inline src/app/components/window-terminal/window-terminal.component.ts]: please use single quotes. also, I would named label `Save Screen Buffer` so users get an idea what actually this is.

File: src/app/components/window-terminal/window-terminal.component.ts [hunk 2]
```diff
@@ -80,6 +84,28 @@ export class WindowTerminalComponent implements OnInit, OnDestroy {
...
+    
+    [].forEach.call(lines, (line :Node) => { 
+      if (line.textContent.length)
+        lineBuffer.push(line.textContent)
+    });
+
+    dialog.showSaveDialog(remote.getCurrentWindow(), (fn) => {
+      if (!fn) return;
+
+      writeFile(fn, lineBuffer.join(EOL), (err) => {
```
Comment (jkuri) [inline src/app/components/window-terminal/window-terminal.component.ts]: why do you want to save screen buffer to file?

File: src/app/components/window-terminal/window-terminal.component.ts [hunk 3]
```diff
@@ -80,6 +84,28 @@ export class WindowTerminalComponent implements OnInit, OnDestroy {
...
+  saveBuffer(): void {
+    let parent: HTMLIFrameElement = <HTMLIFrameElement>document.querySelector('.terminal-instance.active iframe');
+    if (!parent) return;
+    
+    let doc: HTMLDocument = parent.contentWindow.document;
+    let lines: NodeList = doc.querySelectorAll('x-row');
+    let lineBuffer: string[] = [];
+    
+    [].forEach.call(lines, (line :Node) => { 
+      if (line.textContent.length)
```
Comment (jkuri) [inline src/app/components/window-terminal/window-terminal.component.ts]: use brackets here please. I need to add tslint asap (I know).

File: src/app/components/window-terminal/window-terminal.component.ts (hunk 2 above)
Comment (marhi) [inline src/app/components/window-terminal/window-terminal.component.ts]: Sometimes you write a lot of different commands to achive something and then you might want to create a shell script out of that or show somebody the procedure...

Comment (jkuri) [inline src/app/components/window-terminal/window-terminal.component.ts]: okay got it. maybe this could be optional?

Comment (marhi) [inline src/app/components/window-terminal/window-terminal.component.ts]: Ok, I will make it a config option.

File: src/app/components/window-terminal/window-terminal.component.ts [hunk 4]
```diff
@@ -74,13 +78,36 @@ export class WindowTerminalComponent implements OnInit, OnDestroy {
...
     this.hterm.terminals[this.hterm.currentIndex].output.emit(clipboard.readText());
   }
 
+  saveBuffer(): void {
+    let parent: HTMLIFrameElement = <HTMLIFrameElement>document.querySelector('.terminal-instance.active iframe');
+    if (!parent) { return; }
+
+    let doc: HTMLDocument = parent.contentWindow.document;
+    let lines: NodeList = doc.querySelectorAll('x-row');
+    let lineBuffer: string[] = [];
```
Comment (jkuri) [inline src/app/components/window-terminal/window-terminal.component.ts]:
```ts
let lineBuffer: string[] = [].filter.call(lines, (line: Node) => line.textContent.length);
```
is shorter.

**Pull request closed without merging**